### PR TITLE
Missing plot fix

### LINKED ISSUE #28: Missing plots when there are multiple categories
There was some confusion with the flags.  If there were multiple categories, this would prevent the boxplot figure to save.  First noticed in this [post](https://forum.qiime2.org/t/gneiss-balances-boxplot-doesnt-show-up/2169)

### PR DESCRIPTION
**Pull request by mortonjt**
This fixes #28 .  Updating the test cases to sanity check that the plots are actually generated.

### CODE REVIEW

**Comment by mortonjt** [comment]
Looks like something weird is going on with conda

**Comment by mortonjt** [comment]
@ebolyen is this conda issue identified [here](https://forum.qiime2.org/t/new-conda-installation-scheme/2528)?

**Comment by ebolyen** [comment]
@mortonjt, yup, we haven't gone through to update travis to use the new instructions yet. We'll probably get that sorted out sometime today.

**Comment by ebolyen** [comment]
@mortonjt, travis.yml has been updated, but you might need to rebase this PR

**Comment by mortonjt** [comment]
PR rebased @ebolyen thanks!

**Comment by jairideout** [comment]
@mortonjt, sorry for the trouble, can you please rebase and fix the merge conflicts? This code was updated as part of the Metadata overhaul in #32.

**Comment by mortonjt** [comment]
Not a problem.  Just rebased.  Let's see if the tests are still ok.

**Comment by jairideout** [comment]
Let me know if you need help debugging this -- I'm not sure what you're going for in the current PR diff, looks like a partial revert to the pre-Metadata overhaul?

**Comment by mortonjt** [comment]
Things got a little mixed up during the rebasing :/ 
Investigating ...

**Comment by mortonjt** [comment]
Looks like the tests are passing.  Not sure why the coverage went down (we added more tests).

File: q2_gneiss/plot/_plot.py [hunk 1]
```diff
@@ -104,15 +104,17 @@ def f(x):
...
     if metadata is not None:
         fig2, ax = plt.subplots()
         c = metadata.to_series()
         data, c = match(balances, c)
         data[c.name] = c
         y = data[balance_name]
 
+        # check if continuous
         if isinstance(metadata, qiime2.NumericMetadataColumn):
+            c = c.astype(np.float64)
```
Comment (jairideout) [inline q2_gneiss/plot/_plot.py]: Type-casting is no longer necessary with the new Metadata objects -- `NumericMetadataColumn` has the data stored as floating point values.

File: q2_gneiss/plot/_plot.py [hunk 2]
```diff
@@ -124,7 +126,13 @@ def f(x):
             )
             sample_palette = pd.Series(sns.color_palette("Set2", 2),
                                        index=dcat.value_counts().index)
+
         elif isinstance(metadata, qiime2.CategoricalMetadataColumn):
+            if threshold is not None:
```
Comment (jairideout) [inline q2_gneiss/plot/_plot.py]: When I updated the plugin to use the new Metadata API, I moved this check earlier in the method (see [this line](https://github.com/qiime2/q2-gneiss/pull/29/files#diff-9df90a58a11c42324dad341ada319215R40)). Removing the check here will probably fix the drop in coverage.

**Review by jairideout — CHANGES_REQUESTED** [review]
Thanks @mortonjt! I had a couple of comments related to the new Metadata API (one of the comments should help you resolve the coverage drop you're seeing).

File: q2_gneiss/plot/_plot.py (hunk 1 above)
Comment (mortonjt) [reply q2_gneiss/plot/_plot.py]: That's awesome!

**Comment by mortonjt** [comment]
Thanks @jairideout for looking over this!

**Review by jairideout — APPROVED** [review]
Looks great, thanks @mortonjt!

**Comment by mortonjt** [comment]
Thanks guys!

**Pull request merged**